soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -58,59 +113,64 @@ const pll_setup_t pll0Setup = {
 
 static ALWAYS_INLINE void clock_init(void)
 {
-
-#if defined(CONFIG_SOC_LPC55S36)
-	/* Power Management Controller initialization */
+#if (CONFIG_SOC_LPC55S36)
+	/* Power Management Controller Initialization */
 	POWER_PowerInit();
 #endif
-
-#if defined(CONFIG_SOC_LPC55S06) || defined(CONFIG_SOC_LPC55S16) || \
-	defined(CONFIG_SOC_LPC55S28) || defined(CONFIG_SOC_LPC55S36) || \
-	defined(CONFIG_SOC_LPC55S69_CPU0)
     /*!< Set up the clock sources */
     /*!< Configure FRO192M */
 	/*!< Ensure FRO is on  */
 	POWER_DisablePD(kPDRUNCFG_PD_FRO192M);
 	/*!< Set up FRO to the 12 MHz, just for sure */
-	CLOCK_SetupFROClocking(12000000U);
+	CLOCK_SetupFROClocking(12000000);
 	/*!< Switch to FRO 12MHz first to ensure we can change the clock */
 	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
 
-	/* Enable FRO HF(96MHz) output */
-	CLOCK_SetupFROClocking(96000000U);
+#if defined(CONFIG_INIT_PLL0) && !defined(CONFIG_SOC_LPC55S36)
```

Comment:
why does enabling XTAL32M depend on using PLL0 or which SOC is used?  Users may want this oscillator enabled, even if not using PLL0.